### DISCUSSION TITLE
Add searchable flag on Field configure panel

### DISCUSSION THREAD

**Discussion by imiskolee** [directus/directus#12250] [Feature Requests]
hi:

I found the common search text input always searches all fields, including date time/number type fields, it's very slow when we working on a big table.

can we add a boolean flag in the field configure panel(default as true).

**Comment by rijkvanzanten** [comment]
Hmm good one! I'd normally argue that you might want to use the regular filters to pick and choose what fields to search in, but that can get a little cumbersome if you still want to search 10+ fields 🤔

**Reply by imiskolee** [reply]
regular filters are too hard for normal users :), and another reason related to SQL performances.

**Reply by benhaynes** [reply]
I actually like what @bamboowonder recommended over here: https://github.com/directus/directus/discussions/13172

Instead of further cluttering the Field Drawer, we add a "Search" section to the collection with easy toggles. This would also (maybe?) allow us to include some relational fields as well... which is sorely missing.

**Reply by bamboowonder** [reply]
> regular filters are too hard for normal users :), and another reason related to SQL performances.

I agree.  In our case, the table has about 20 columns. and about 200 rows.   I want the "search" to ONLY search on ONE field.  The "company_name".  But instead it just does nothing and end users have to create a filter just for that.   Kind've defeats the purpose of even having a search box?

**Reply by errnesto** [reply]
Just came here to second this. I think what to search and what not really depends on the data model.

With our data model the search is basically useless. 
Because one entry has many fields and something always matches the search string. 
So the results are very unpredictable. 

And since relations are not included in the search this makes it even harder for users who don't have a concept of which fields are relations and which are not.

But yeah even if we could only chosse a single field that is the "searchable field" that would completely solve this :-)

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
